File: src/main/java/calebzhou/rdi/craftsphere/misc/ServerConnector.java
package calebzhou.rdi.craftsphere.misc;

import calebzhou.rdi.craftsphere.RdiCore;
import calebzhou.rdi.craftsphere.screen.RdiTitleScreen;
import calebzhou.rdi.craftsphere.util.ThreadPool;
import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.screens.ConnectScreen;
import net.minecraft.client.multiplayer.ServerData;
import net.minecraft.client.multiplayer.ServerStatusPinger;
import net.minecraft.client.multiplayer.resolver.ServerAddress;

public class ServerConnector {
    public static final ServerStatusPinger PINGER = new ServerStatusPinger();
    public static final ServerAddress SERVER_ADDRESS = RdiCore.debug?new ServerAddress("localhost",25565):new ServerAddress("test3.davisoft.cn",26085);
    public static final ServerData SERVER_INFO = new ServerData(RdiCore.MODID_CHN+ RdiCore.VER_DISPLAY, SERVER_ADDRESS.getHost()+":"+SERVER_ADDRESS.getPort(),false);
    public static void connect(){
        Minecraft.getInstance().getWindow().setTitle(RdiCore.MODID_CHN+" "+ RdiCore.VER_DISPLAY);
        ConnectScreen.startConnecting(RdiTitleScreen.INSTANCE, Minecraft.getInstance(), SERVER_ADDRESS,SERVER_INFO);
    }
    public static void ping(){
        ThreadPool.newThread(()->{
            try {
                PINGER.pingServer(SERVER_INFO, () -> {});
            } catch (Exception e) {
                e.printStackTrace();
            }
        });

    }
}

File: src/main/java/calebxzhou/rdi/misc/ServerConnector.kt
package calebxzhou.rdi.misc

import calebxzhou.libertorch.util.ThreadPool
import calebxzhou.rdi.consts.RdiConsts.CoreVersion
import calebxzhou.rdi.consts.RdiConsts.DEBUG
import calebxzhou.rdi.consts.RdiConsts.MODID_DISPLAY
import calebxzhou.rdi.screen.RdiTitleScreen
import net.minecraft.client.Minecraft
import net.minecraft.client.gui.screens.ConnectScreen
import net.minecraft.client.multiplayer.ServerData
import net.minecraft.client.multiplayer.ServerStatusPinger
import net.minecraft.client.multiplayer.resolver.ServerAddress

object ServerConnector {
    val PINGER = ServerStatusPinger()
    val SERVER_ADDRESS =
        if (DEBUG) ServerAddress("localhost", 26085) else ServerAddress("test3.davisoft.cn", 26085)
    val SERVER_INFO = ServerData(
        MODID_DISPLAY + CoreVersion,
        SERVER_ADDRESS.host + ":" + SERVER_ADDRESS.port,
        false
    )

    @JvmStatic
	fun connect() {
        Minecraft.getInstance().window.setTitle("$MODID_DISPLAY $CoreVersion")
        ConnectScreen.startConnecting(RdiTitleScreen(), Minecraft.getInstance(), SERVER_ADDRESS, SERVER_INFO)

    }

    fun ping() {
        ThreadPool.run() {
            try {
                PINGER.pingServer(SERVER_INFO) {}
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
    }
}
